presto-spi/src/main/java/com/facebook/presto/spi/block/MapBlockBuilder.java

```diff
@@ -70,6 +70,31 @@ public MapBlockBuilder(
                 newNegativeOneFilledArray(expectedEntries * HASH_MULTIPLIER));
     }
 
+    // Caller of this constructor is responsible for making sure `keyBlockBuilder` and `valueBlockBuilder`
+    // is constructed with the same `blockBuilderStatus` as the one in the argument
+    public MapBlockBuilder(
```

Comment:
The reason for doing this is to have control of the capacity of underlying block builders. However, I am actually not a big fan of this constructor...

Reply:
I don't like this either. It feels like to me that you created this constructor so that you can test it. However, there wouldn't be a need to test it if it didn't exist.